[Dinoman44] iP

Dinoman44 opened the pull request:
# Fret
`Fret`, named after the player support AI [Fret from Halo Infinite](https://www.halopedia.org/FRET), is a chatbot that will take note of your tasks and allow you to mark, delete, save tasks and also add other tasks as needed.

The tone of the bot when you give it commands mimics its voice lines in-game and suits its personality well:
> "Fret AI constructs are able to remain eager to tackle the task ahead"

## Capabilities
The bot is able to undertake the following tasks:
 * Adding tasks
 * Deleting tasks
 * Listing all tasks along with completion status
 * Marking tasks as complete/incomplete
 * Saving tasks to memory and reading from them when restarted

To do any of the above, just follow these steps:
1. Start the bot
2. Enter a command
3. Done! 🗿

Features scheduled for later release:
 - [X] ~~Handling of proper dates and times for deadlines/events~~ (**RELEASED**)
 - [ ] Save tasks to any specified text file

<hr>

Want to contribute? Read on:
```java
public class Contributer {
    public void contribute(Person p) {
        Email.sendMessage(p, "Dinoman44", "I want to help out!");
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Dinoman44 committed:
Add files via upload

Dinoman44 committed:
Complete Level-1 requirements

Dinoman44 committed:
Delete src/main/java/Duke.java

Dinoman44 committed:
Add task tracker for Level 2

Dinoman44 committed:
Add feature: marking/unmarking task completion

Dinoman44 committed:
Add handling for different task types

Dinoman44 committed:
Add task deletion functionality

Dinoman44 committed:
Level-4 commit

Dinoman44 committed:
UI Testing

Dinoman44 committed:
Read tasks from memory

Dinoman44 committed:
Store tasks in memory

Dinoman44 committed:
Allow OS-independant file pathing

Dinoman44 committed:
Add proper datetime functionality and handling

Dinoman44 committed:
Break up Fret functionality into multiple classes

Dinoman44 committed:
Separate new classes into packages (task and command)

Dinoman44 committed:
Edit gitignore

Dinoman44 committed:
Add Gradle

Dinoman44 committed:
Add JUnit testing for some classes

Dinoman44 committed:
Allow Jar file packaging

Dinoman44 committed:
Add JavaDoc comments to all non-trivial methods

Dinoman44 committed:
Add annotations and order imports to comply with coding standards

Dinoman44 committed:
Add search functionality

Dinoman44 committed:
Edit methods and classes to comply with coding standards

Dinoman44 committed:
Add search functionality

Dinoman44 committed:
Update gradle build to include JavaFX dependencies

Dinoman44 committed:
Refactor code to package Parser, Ui and Storage classes

Dinoman44 committed:
Add checkstyle

huangjiaxi1111 commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,52 @@
+
+public abstract class Command {
+    private final Random RNG = new Random();
+    private int i;
```
Perhaps you could rename `i` to a more descriptive name since it has large scope

Dinoman44 replied:
That makes sense, I'll rename it to `taskIndex` since that's what it refers to.

huangjiaxi1111 commented on src/main/java/psu/Parser.java:
```diff
@@ -0,0 +1,181 @@
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import command.*;
```
Perhaps you should list all imports :(

Dinoman44 replied:
Yeah I was lazy 🫣

huangjiaxi1111 commented on src/main/java/psu/Parser.java:
```diff
@@ -0,0 +1,181 @@
+    private static final Pattern TODO_PATTERN = Pattern.compile("todo (.+)");
+    private static final Pattern DEADLINE_PATTERN = Pattern.compile("deadline (.+?) /by (.+)");
+    private static final Pattern EVENT_PATTERN = Pattern.compile("event (.+?) /from (.+) /to (.+)");
+    private static final Pattern DATETIME_PATTERN = Pattern.compile("(\\d+)[ \\/\\\\\\|\\-\\_](\\d+)[ \\/\\\\\\|\\-\\_](\\d+)[ tT]([\\d:]+) *(am|pm)");
```
Perhaps you should wrap this line :(

Dinoman44 replied:
Will do, thanks!

huangjiaxi1111 commented on src/test/java/DeadlineTest.java:
```diff
@@ -0,0 +1,39 @@
+
+public class DeadlineTest {
+    @Test
+    public void test1() {
```
Perhaps test method name can be more descriptive

Dinoman44 replied:
Fair point, thanks alot!

huangjiaxi1111 reviewed (commented):
Your code is very well structured and abides by the naming convention well! Just need to fix some minor issues mentioned below.

Dinoman44 commented:
Thanks for the review, I'll work on your suggestions!

Dinoman44 committed:
Add GUI using JavaFX

Dinoman44 committed:
Adjust code based on PR review 1 feedback

AbdulrahmanAlRammah commented on src/main/java/fret/Fret.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    /**
+     * Runs the Fret chatbot program
```
Should there be a period at the end of Javadoc lines?

Dinoman44 replied:
Oh yes you are right. Thanks for pointing it out!

AbdulrahmanAlRammah commented on src/main/java/command/TaskCompletionCommand.java:
```diff
@@ -0,0 +1,54 @@
+
+public class TaskCompletionCommand extends Command {
+    private boolean isCompleted;
+    private final String[] TASK_COMPLETED_PREFIXES = new String[] {
```
Can this be static?

Dinoman44 replied:
Hmm yes probably.

AbdulrahmanAlRammah commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,39 @@
+import java.time.LocalDateTime;
+
+public class Event extends Task {
+    LocalDateTime startDate;
```
Could this be protected or private?

Dinoman44 replied:
It was supposed to be private, seems like I did not write it in correctly.

AbdulrahmanAlRammah commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,39 @@
+
+    /**
+     * Converts the LocalDateTime object to a cleaner looking string.
+     * Similar to Deadline::dateTimeToCleanString
+     * 
+     * @param dt the LocalDateTime object to be converted
+     * @return the date and time formatted like "10:30pm on AUGUST 29, 2024"
```
Should there be periods at the end of these lines?

Dinoman44 replied:
Yes, thanks!

AbdulrahmanAlRammah commented on src/main/java/command/TaskCompletionCommand.java:
```diff
@@ -0,0 +1,54 @@
+
+    private final String[] TASK_UNCOMPLETED_PREFIXES = new String[] {
+        "Marking as incomplete.\n",
+        "Booooo\n",
```
I like this as it makes the chatbot more lively!

Dinoman44 replied:
Thank you!

AbdulrahmanAlRammah reviewed (commented):
Overall well done and well packaged code. Some nit comments added.

Dinoman44 committed:
Adjust code based on PR review 2 feedback

Dinoman44 committed:
Rebuild jar file after accounting for checkstyle

Checkstyle picks up import order and spacing errors in some files. It also notes that some methods are missing JavaDoc comments.

Additionally, the checkstyle files are not picked up by gradle.

This means that some parts of the code have some styling errors, and gradle is unable to check for them when building the jar file.

Let's edit the code in such a way that the checkstyle files are picked up by gradle and styling tests are run upon building the jar file.
Also add JavaDoc comments to more methods and fix the import order and spacing issues present in some files.

This allows the code to adhere to the coding convention specified and prevents the build from failing.

Refer to [this link](https://se-education.org/guides/conventions/java/index.html) for the Java coding standard.

Dinoman44 committed:
Add assertions

Dinoman44 committed:
Remove Ui.java for redundancy

Dinoman44 committed:
task/TaskList.java: Edit search method to use streams

Dinoman44 committed:
Merge pull request #1 from Dinoman44/branch-A-Assertions

Add assertions

Dinoman44 committed:
Conclude merge of branch-A-CodeQuality

Dinoman44 committed:
task/TaskList.java: Edit search to use streams

Dinoman44 committed:
Allow users to update task details

Dinoman44 committed:
Update user guide

Dinoman44 committed:
Add more assertions

Dinoman44 committed:
Update user guide

Dinoman44 committed:
Add more bot response prefixes

Dinoman44 committed:
Update style based on SE script feedback

Dinoman44 committed:
Update docs/README.md

Dinoman44 committed:
Update docs/README.md

Dinoman44 committed:
Add placeholder image to sidebar of chat window

Dinoman44 committed:
Merge branch 'master' of https://github.com/Dinoman44/ip

Reconciling edited docs/README.md file.

Dinoman44 committed:
Update docs/README.md

Dinoman44 committed:
docs/README.md: Add FAQ section

Dinoman44 committed:
Add new help command

Dinoman44 committed:
docs/README.md: Edit update task command section

Dinoman44 committed:
Update GUI (colors, side banner)

Dinoman44 committed:
Add Ui.png
